### ISSUE TITLE
Script filename "App.py" cannot be processed by pyinstaller if a package named "App" resides in the same directory

### ISSUE THREAD

**Issue by kostyanoob** [pyinstaller/pyinstaller#5067]
**Bug description:** 
When your main script is called App.py, and there is a package directory called App, and you apply pyinstaller on App.py, the resulting EXE doesn't run.

**My system:**
Windows 10-64bit, Python 3.8.2, pyinstaller 4.0dev

**The minimal example to reproduce the bug:**
1) **[Edited this item]** Create file named _App.py_ with a single line in it, and create a directory _App_ with an empty \_\__init_\_\_._py_ in it:
`C:\>echo print("This should have been printed") > App.py`
`C:\>mkdir App`
`C:\>type NUL > App\__init__.py`
2) Run it with a python interpreter, and correct behavior is observed.
`C:\>python App.py`
`This should have been printed`
3) Try bundling it with pyinstaller
`C:\>pyinstaller App.py --console`
4) Run the resulting EXE and it will not print (This is the problem).
`C:\>dist\App\App.exe`
`C:\>`
5) Rename _App.py_ to _hello.py_
`C:\>copy App.py hello.py`
6) Re-run the pyinstaller
`C:\>pyinstaller hello.py --console`
7) Run the resulting EXE and it prints (no problem).
`C:\>dist\hello\hello.exe`
`This should have been printed`

**Comment by amifunny** [comment]
@kostyanoob I don't think, `.py` filename should be a problem. I have checked `dev version` on my system with slightly different specs but everything is fine. Can you check in an editor if the content of `App.py ` is what you think it is?

**Comment by bwoodsend** [comment]
Likewise - I'm unable to reproduce on Windows 8.

**Comment by bwoodsend** [comment]
Clutching at straws here but can you check for anything that may also be named _App_? 
To check in `PATH` Use:
```shell
where App
```
And to check for Python modules:
```shell
pip freeze | grep App
```
And check if `import App` raises an `ModuleNotFound` error.

**Comment by kostyanoob** [comment]
Hi fellows, thank you for the responsiveness. Indeed there was one more detail I omitted, which is the existence of the _App_ directory with a _\_\_init\_\_.py_ in it which makes _App_ a package. In the step-by-step reproduction instructions, I now updated "step 1", and I verified again that this reproduces the issue.

I didn't receive any ambiguity warnings about this issue, just an executable that is not running properly. I acknowledge that this is a poor package naming accident on my side. However, maybe it will help you to add some sort of warning about it in the future versions. 

Thanks.
